tls/extensions/s2n_client_key_share.c

```diff
@@ -99,14 +118,7 @@ static int s2n_generate_pq_hybrid_key_share(struct s2n_stuffer *out, struct s2n_
 
     struct s2n_ecc_evp_params *ecc_params = &kem_group_params->ecc_params;
     ecc_params->negotiated_curve = kem_group->curve;
-    POSIX_GUARD(s2n_stuffer_write_uint16(out, ecc_params->negotiated_curve->share_size));
-
-    /* If we received a HRR for any reason other than to request a different key share,
-     * we might have already generated the key. */
-    if (ecc_params->evp_pkey == NULL) {
-        POSIX_GUARD(s2n_ecc_evp_generate_ephemeral_key(ecc_params));
-    }
-    POSIX_GUARD(s2n_ecc_evp_write_params_point(ecc_params, out));
+    POSIX_GUARD_RESULT(s2n_ecdhe_send_public_key(out, ecc_params));
```

Comment:
Agreed. Unfortunately consolidating the classical / PQ logic isn't straightforward. I could probably figure something out (macros at the worst) but it would definitely be a bigger refactor. Believe it or not, I'm trying to keep this change small :)